src/xpub.ts

```diff
@@ -336,6 +338,29 @@ class Xpub extends EventEmitter {
     const inserted = await this.storage.appendTxs(txs);
     return inserted;
   }
+
+  async checkReorg(filter: { account: number; index: number }) {
+    const lastTx = await this.storage.getLastTx(filter);
+
+    if (!lastTx) {
+      return;
+    }
+
+    const block = await this.explorer.getBlockByHeight(lastTx.block.height);
+
+    // all good the block is valid
+    if (block && block.hash === lastTx.block.hash) {
+      return;
+    }
+
+    // in this case the block is not valid so we delete everything
+    // for that block
+
+    await this.storage.removeTxs(filter);
+
+    // keep going until we take the all good exit
+    await this.checkReorg(filter);
```

Comment:
actually there is not need to do recursion here because we delete the whole address history

Reply:
but we can argue that this function is not aware of that